File: DXDAL/Carriers.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Helpers
Imports DevExpress.Data.Filtering

Namespace SPGData
    <Persistent("tblCarriers")>
    Public Class Carriers
        Inherits XPLiteObject
        Dim fCarrierID As Integer
        <Key()> _
        Public Property CarrierID() As Integer
            Get
                Return fCarrierID
            End Get
            Set(ByVal value As Integer)
                SetPropertyValue(Of Integer)("CarrierID", fCarrierID, value)
            End Set
        End Property
        Dim fCarrierName As String
        <Size(50)> _
        Public Property CarrierName() As String
            Get
                Return fCarrierName
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("CarrierName", fCarrierName, value)
            End Set
        End Property
        Dim fContactFirstName As String
        <Size(50)> _
        Public Property ContactFirstName() As String
            Get
                Return fContactFirstName
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("ContactFirstName", fContactFirstName, value)
            End Set
        End Property
        Dim fContactLastName As String
        <Size(50)> _
        Public Property ContactLastName() As String
            Get
                Return fContactLastName
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("ContactLastName", fContactLastName, value)
            End Set
        End Property
        Dim fAddress As String
        <Size(150)> _
        Public Property Address() As String
            Get
                Return fAddress
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Address", fAddress, value)
            End Set
        End Property
        Dim fCity As String
        <Size(50)> _
        Public Property City() As String
            Get
                Return fCity
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("City", fCity, value)
            End Set
        End Property
        Dim fState As String
        <Size(50)> _
        Public Property State() As String
            Get
                Return fState
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("State", fState, value)
            End Set
        End Property
        Dim fPostal As String
        <Size(50)> _
        Public Property Postal() As String
            Get
                Return fPostal
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Postal", fPostal, value)
            End Set
        End Property
        Dim fPhone As String
        <Size(50)> _
        Public Property Phone() As String
            Get
                Return fPhone
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Phone", fPhone, value)
            End Set
        End Property
        Dim fEmail As String
        <Size(50)> _
        Public Property Email() As String
            Get
                Return fEmail
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Email", fEmail, value)
            End Set
        End Property
        Dim fFax As String
        <Size(50)> _
        Public Property Fax() As String
            Get
                Return fFax
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Fax", fFax, value)
            End Set
        End Property
        Dim fNote As String
        <Size(50)> _
        Public Property Note() As String
            Get
                Return fNote
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Note", fNote, value)
            End Set
        End Property
        Dim fEnteredBy As String
        <Size(50)> _
        Public Property EnteredBy() As String
            Get
                Return fEnteredBy
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("EnteredBy", fEnteredBy, value)
            End Set
        End Property
        Dim fEnteredOn As DateTime
        Public Property EnteredOn() As DateTime
            Get
                Return fEnteredOn
            End Get
            Set(ByVal value As DateTime)
                SetPropertyValue(Of DateTime)("EnteredOn", fEnteredOn, value)
            End Set
        End Property
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Public Sub New()
            MyBase.New(Session.DefaultSession)
        End Sub
        Public Overrides Sub AfterConstruction()
            MyBase.AfterConstruction()
        End Sub

        Public Shadows Class FieldsClass
            Inherits XPLiteObject.FieldsClass
            Public Sub New()
                MyBase.New()
            End Sub
            Public Sub New(ByVal propertyName As String)
                MyBase.New(propertyName)
            End Sub
            Public ReadOnly Property CarrierID() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CarrierID"))
                End Get
            End Property
            Public ReadOnly Property CarrierName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("CarrierName"))
                End Get
            End Property
            Public ReadOnly Property ContactFirstName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ContactFirstName"))
                End Get
            End Property
            Public ReadOnly Property ContactLastName() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("ContactLastName"))
                End Get
            End Property
            Public ReadOnly Property Address() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Address"))
                End Get
            End Property
            Public ReadOnly Property City() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("City"))
                End Get
            End Property
            Public ReadOnly Property State() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("State"))
                End Get
            End Property
            Public ReadOnly Property Postal() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Postal"))
                End Get
            End Property
            Public ReadOnly Property Phone() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Phone"))
                End Get
            End Property
            Public ReadOnly Property Email() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Email"))
                End Get
            End Property
            Public ReadOnly Property Fax() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Fax"))
                End Get
            End Property
            Public ReadOnly Property Note() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("Note"))
                End Get
            End Property
            Public ReadOnly Property EnteredBy() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("EnteredBy"))
                End Get
            End Property
            Public ReadOnly Property EnteredOn() As OperandProperty
                Get
                    Return New OperandProperty(GetNestedName("EnteredOn"))
                End Get
            End Property
        End Class
        Private Shared _fields As FieldsClass
        Public Shared Shadows ReadOnly Property Fields() As FieldsClass
            Get
                If ReferenceEquals(_fields, Nothing) Then
                    _fields = New FieldsClass()
                End If
                Return _fields
            End Get
        End Property

    End Class
End Namespace

File: DXDAL/Carriers.cs
using System;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace DXDAL
{
    namespace SPGData
    {
        [Persistent("tblCarriers")]
        public class Carriers : XPLiteObject
        {
            private int fCarrierID;
            [Key()]
            public int CarrierID
            {
                get => fCarrierID;
                set => SetPropertyValue<int>("CarrierID", ref fCarrierID, value);
            }
            private string fCarrierName;
            [Size(50)]
            public string CarrierName
            {
                get => fCarrierName;
                set => SetPropertyValue("CarrierName", ref fCarrierName, value);
            }
            private string fContactFirstName;
            [Size(50)]
            public string ContactFirstName
            {
                get => fContactFirstName;
                set => SetPropertyValue("ContactFirstName", ref fContactFirstName, value);
            }
            private string fContactLastName;
            [Size(50)]
            public string ContactLastName
            {
                get => fContactLastName;
                set => SetPropertyValue("ContactLastName", ref fContactLastName, value);
            }
            private string fAddress;
            [Size(150)]
            public string Address
            {
                get => fAddress;
                set => SetPropertyValue("Address", ref fAddress, value);
            }
            private string fCity;
            [Size(50)]
            public string City
            {
                get => fCity;
                set => SetPropertyValue("City", ref fCity, value);
            }
            private string fState;
            [Size(50)]
            public string State
            {
                get => fState;
                set => SetPropertyValue("State", ref fState, value);
            }
            private string fPostal;
            [Size(50)]
            public string Postal
            {
                get => fPostal;
                set => SetPropertyValue("Postal", ref fPostal, value);
            }
            private string fPhone;
            [Size(50)]
            public string Phone
            {
                get => fPhone;
                set => SetPropertyValue("Phone", ref fPhone, value);
            }
            private string fEmail;
            [Size(50)]
            public string Email
            {
                get => fEmail;
                set => SetPropertyValue("Email", ref fEmail, value);
            }
            private string fFax;
            [Size(50)]
            public string Fax
            {
                get => fFax;
                set => SetPropertyValue("Fax", ref fFax, value);
            }
            private string fNote;
            [Size(50)]
            public string Note
            {
                get => fNote;
                set => SetPropertyValue("Note", ref fNote, value);
            }
            private string fEnteredBy;
            [Size(50)]
            public string EnteredBy
            {
                get => fEnteredBy;
                set => SetPropertyValue("EnteredBy", ref fEnteredBy, value);
            }
            private DateTime fEnteredOn;
            public DateTime EnteredOn
            {
                get => fEnteredOn;
                set => SetPropertyValue<DateTime>("EnteredOn", ref fEnteredOn, value);
            }
            public Carriers(Session session) : base(session)
            {
            }
            public Carriers() : base(Session.DefaultSession)
            {
            }
            public override void AfterConstruction()
            {
                base.AfterConstruction();
            }

            public new class FieldsClass : PersistentBase.FieldsClass
            {
                public FieldsClass() : base()
                {
                }
                public FieldsClass(string propertyName) : base(propertyName)
                {
                }
                public OperandProperty CarrierID => new OperandProperty(GetNestedName("CarrierID"));
                public OperandProperty CarrierName => new OperandProperty(GetNestedName("CarrierName"));
                public OperandProperty ContactFirstName => new OperandProperty(GetNestedName("ContactFirstName"));
                public OperandProperty ContactLastName => new OperandProperty(GetNestedName("ContactLastName"));
                public OperandProperty Address => new OperandProperty(GetNestedName("Address"));
                public OperandProperty City => new OperandProperty(GetNestedName("City"));
                public OperandProperty State => new OperandProperty(GetNestedName("State"));
                public OperandProperty Postal => new OperandProperty(GetNestedName("Postal"));
                public OperandProperty Phone => new OperandProperty(GetNestedName("Phone"));
                public OperandProperty Email => new OperandProperty(GetNestedName("Email"));
                public OperandProperty Fax => new OperandProperty(GetNestedName("Fax"));
                public OperandProperty Note => new OperandProperty(GetNestedName("Note"));
                public OperandProperty EnteredBy => new OperandProperty(GetNestedName("EnteredBy"));
                public OperandProperty EnteredOn => new OperandProperty(GetNestedName("EnteredOn"));
            }
            private static FieldsClass _fields;
            public static new FieldsClass Fields
            {
                get
                {
                    if (ReferenceEquals(_fields, null))
                        _fields = new FieldsClass();
                    return _fields;
                }
            }
        }
    }
}
